### PR TITLE
Let builder-nodejs use node 11.x

### PR DESCRIPTION
**Pull request by sboardwell**

### CODE REVIEW

**Comment by scherler** [comment]
We need to pin down the npm version and not blindly use the latest :bug:

**Comment by mkramb** [comment]
@sboardwell 
- what is the issue you are referring to #4656, can provide a link
- we need to ping the NPM version, I think this change will always upgrade `npm` to latest

**Comment by sboardwell** [comment]
@mkramb / @scherler : it's a good point. I was just going by the command `npm -i -g watch-cli vsce typescript`, although I agree that npm is a more powerful/important package and should be maanged. I'll add a version now.

File: builder-nodejs/Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@ FROM jenkinsxio/builder-base:1
 RUN curl -f --silent --location https://rpm.nodesource.com/setup_9.x | bash - && \
```
Comment (scherler) [inline builder-nodejs/Dockerfile]: `https://rpm.nodesource.com/setup_9.x` should be replaced with `https://rpm.nodesource.com/setup_11.x`

Comment (sboardwell) [reply builder-nodejs/Dockerfile]: 👍

File: builder-nodejs8x/Dockerfile [hunk 2]
```diff
@@ -3,7 +3,7 @@ FROM jenkinsxio/builder-base:1
 RUN curl -f --silent --location https://rpm.nodesource.com/setup_8.x | bash - && \
   yum install -y nodejs gcc-c++ make bzip2 GConf2 gtk2 chromedriver chromium xorg-x11-server-Xvfb
 
-RUN npm i -g watch-cli vsce typescript && npm up -g npm
+RUN npm i -g watch-cli vsce typescript npm@6.5.0 && [ "6.5.0" == "$(npm -v)" ]
```
Comment (scherler) [inline builder-nodejs8x/Dockerfile]: I think you can drop the ` && [ "6.5.0" == "$(npm -v)" ]` part, since that is always true

**Comment by mkramb** [comment]
@sboardwell 
Why is this even being updated?
Can you provide a link to the original issue?

**Comment by scherler** [comment]
https://issues.jenkins-ci.org/browse/JENKINS-4656 is not the issue I guess, can you provide a link to understand the reasons behind the change? TIA

**Comment by sboardwell** [comment]
Oh, sorry, I though I had provided this: https://github.com/jenkins-x/jenkins-x-platform/issues/4656

**Comment by mkramb** [comment]
@sboardwell 
To upgrade node in the base image, that's fine.

But by installing node e.g. `https://rpm.nodesource.com/setup_11.x`, that comes with an associated `npm` version, and I don't think we should manuall upgrade / lock `npm` to a version. If that is required it shouldn't be done in the base image.

**Comment by sboardwell** [comment]
Oh mann... 😄

@scherler / @mkramb: scratch the npm install changes. If it's okay with you I would still like to update the builder-nodejs to 11.x though since 9.x is EOM.

Re: the reason for my npm install changes

I thought that node8.x and even node10.x were still shipping with npm 5.6.0. 

It turns out I had a problem with my pipeline 🤦‍♂️ , or more to the point, the yaml override. 

Here's the broken pipeline:
```
pipeline {
    agent {
        kubernetes {
            inheritFrom 'nodejs'
            defaultContainer 'nodejs'
            yaml '''
apiVersion: v1
kind: Pod
metadata:
  labels:
    override-key: override-value
spec:
  containers:
  - name: nodejs
     image: jenkinsxio/builder-nodejs8x:0.1.187
'''
        }
    }
    stages {
        stage('Npm Check') {
            steps {
                sh 'npm -v'
            }
        }
    }
    post {
        always {
            cleanWs()
        }
    }
}
```

I found the override-key in the labels of the created pod but the image override isn't being taken and the old image was being used. Since it didn't through an exception I assumed I was correct.

I will ask in the Slack channels. I think this issue is already known. I just thought it only affected Cpu Resources and such.

Sorry for the run around. I will revert all my changes except the 9.x -> 11.x and edit the title.

**Comment by sboardwell** [comment]
So, just in case, the solution to use the deprecated `containerTemplate` until yaml plays nicely with inheritance. The working Jenkinsfile looks like this:
```
pipeline {
    agent {
        kubernetes {
            inheritFrom 'nodejs'
            label 'my-agent' // mandatory otherwise build hangs
            containerTemplate {
                name 'nodejs'
                image 'jenkinsxio/builder-nodejs8x:0.1.187'
            }
            defaultContainer 'nodejs'
        }
    }
    stages {
        stage('Npm Check') {
            steps {
                sh 'npm -v'
            }
        }
    }
    post {
        always {
            cleanWs()
        }
    }
}
```

**Comment by sboardwell** [comment]
@scherler / @mkramb : can someone check this please? Or should I close this and create a new PR to avoid confusion?

**Pull request merged**